cmd/tailscale/cli/update.go

```diff
@@ -705,3 +782,32 @@ func (pw *progressWriter) print() {
 	pw.lastPrint = time.Now()
 	log.Printf("Downloaded %v/%v (%.1f%%)", pw.done, pw.total, float64(pw.done)/float64(pw.total)*100)
 }
+
+func haveExecutable(name string) bool {
+	path, err := exec.LookPath(name)
+	return err == nil && path != ""
+}
+
+func latestTailscaleVersion(track string) (string, error) {
+	res, err := http.Get("https://pkgs.tailscale.com/" + track + "/?mode=json")
+	if err != nil {
+		return "", fmt.Errorf("fetching latest tailscale version: %w", err)
+	}
+	var latest struct {
+		Tarballs map[string]string // ~goarch (ignoring "geode") => "tailscale_1.34.2_mips.tgz"
```

Comment:
This is now duplicated, too.

Reply:
Updated this to get the `Version` field from the JSON response (after tweaking pkgserve to always return it) and deduped everywhere